x/metadata/keeper/msg_server.go

```diff
@@ -368,6 +368,83 @@ func (k msgServer) DeleteContractSpecification(
 	return types.NewMsgDeleteContractSpecificationResponse(), nil
 }
 
+func (k msgServer) AddContractSpecToScopeSpec(
+	goCtx context.Context,
+	msg *types.MsgAddContractSpecToScopeSpecRequest,
+) (*types.MsgAddContractSpecToScopeSpecResponse, error) {
+	defer telemetry.MeasureSince(time.Now(), types.ModuleName, "tx", "AddContractSpecToScopeSpec")
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	contractSpec, found := k.GetContractSpecification(ctx, msg.ContractSpecificationId)
+	if !found {
+		return nil, fmt.Errorf("contract specification not found with id %s", msg.ContractSpecificationId)
+	}
+	if err := k.ValidateAllOwnersAreSigners(contractSpec.OwnerAddresses, msg.Signers); err != nil {
+		return nil, err
+	}
+	scopeSpec, found := k.GetScopeSpecification(ctx, msg.ScopeSpecificationId)
+	if !found {
+		return nil, fmt.Errorf("scope specification not found with id %s", msg.ScopeSpecificationId)
+	}
+	if err := k.ValidateAllOwnersAreSigners(scopeSpec.OwnerAddresses, msg.Signers); err != nil {
+		return nil, err
+	}
+
+	for _, cSpecID := range scopeSpec.ContractSpecIds {
+		if cSpecID.Equals(msg.ContractSpecificationId) {
+			return nil, fmt.Errorf("scope spec %s already contains contract spec %s", msg.ScopeSpecificationId, msg.ContractSpecificationId)
+		}
+	}
+
+	scopeSpec.ContractSpecIds = append(scopeSpec.ContractSpecIds, msg.ContractSpecificationId)
+	k.SetScopeSpecification(ctx, scopeSpec)
+
+	k.EmitEvent(ctx, types.NewEventTxCompleted(types.TxEndpoint_AddContractSpecToScopeSpec, msg.GetSigners()))
+	return types.NewMsgAddContractSpecToScopeSpecResponse(), nil
+}
+
+func (k msgServer) DeleteContractSpecFromScopeSpec(
+	goCtx context.Context,
+	msg *types.MsgDeleteContractSpecFromScopeSpecRequest,
+) (*types.MsgDeleteContractSpecFromScopeSpecResponse, error) {
+	defer telemetry.MeasureSince(time.Now(), types.ModuleName, "tx", "DeleteContractSpecFromScopeSpec")
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	contractSpec, found := k.GetContractSpecification(ctx, msg.ContractSpecificationId)
+	if !found {
+		return nil, fmt.Errorf("contract specification not found with id %s", msg.ContractSpecificationId)
+	}
+	if err := k.ValidateAllOwnersAreSigners(contractSpec.OwnerAddresses, msg.Signers); err != nil {
```

Comment:
contract spec is not being modified here so its owners are not required to sign this update.